### ISSUE TITLE
tfjs-vis not compatible with tfjs 3.1.0 and higher

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#4746]
regression in tfjs 3.1.0 cannot create sequential model

when importing tfjs using
```js
import * as tf from '@tensorflow/tfjs';
```
and running through a bundler (`esbuild`)  
it fails to add layer to sequential model:
```js
const model = tf.sequential({ name: 'testModel' });
model.add(tf.layers.dense(layerDense);
```

error backtrace using tfjs 3.2.0 (error is same in tfjs 3.1.0 but line numbers may change):
```log
engine.js:699 Uncaught (in promise) TypeError: e.backend.decComplexRef is not a function
    at b.disposeTensor (engine.js:699)
    at x.dispose (tensor.js:283)
    at Dense.addWeight (topology.ts:1298)
    at Dense.build (core.ts:244)
    at topology.ts:993
    at nameScope (common.ts:48)
    at Dense.apply (topology.ts:979)
    at Sequential.add (models.ts:478)
    at createRNN (model.js:47) <-- this is where model.add is called
```

environment: tfjs 3.1.0 or 3.2.0 on chrome 88
reproduction code: <https://github.com/vladmandic/stocks>
error is same regardless of backend used (cpu, webgl, wasm)

now, i've tried tfjs 3.0.0 and it WORKS without issues  

upgrading to tfjs 3.1.0 causes breaking issue which stays the same in tfjs 3.2.0  

however, when using tfjs via script tag in html instead of import in js, everything works (even with tfjs 3.2.0):
```html
<script src="../node_modules/@tensorflow/tfjs/dist/tf.es2017.js"></script>
```

all this likely indicates that some dropped import between tfjs 3.0.0 and tfjs 3.1.0 causes incorrect tree shaking.

only hit is <https://github.com/tensorflow/tfjs/issues/4609>, but that was closed as stale and never resolved.

**Comment by lina128** [comment]
Hi @vladmandic , decComplexRef is removed in 3.2.0. A quick search of the repo shows no occurence of it in our codebase. Can you rm -rf dist node_modules yarn.lock and yarn install everything again? It's most likely cached install.

**Comment by vladmandic** [comment]
ah, found it: latest `tfjs-vis` is still built against older `tfjs`,  
so if its loaded it causes tfjs version conflict and results in this error  

changing title to `tfjs-vis not compatible with tfjs 3.1.0 and higher`

```log
rm -rf node_modules/ package-lock.json ~/.npm/_cacache/
npm install
grep @tensorflow package.json
    "@tensorflow/tfjs-vis": "^1.5.0",
    "@tensorflow/tfjs": "=3.2.0",
    "@tensorflow/tfjs-backend-wasm": "=3.2.0",
    "@tensorflow/tfjs-backend-webgl": "=3.2.0",
grep -Rl decComplexRef node_modules/@tensorflow/
  node_modules/@tensorflow/tfjs-vis/dist/tfjs-vis.umd.min.js
  node_modules/@tensorflow/tfjs-vis/dist/tfjs-vis.umd.min.js.map
```

**Comment by vladmandic** [comment]
just to confirm, tfjs-vis 3.3.0 is still BROKEN - it's built against tfjs 3.0.0:

this should be a trivial fix, why the delay?

```log
 grep @tensorflow package.json
    "@tensorflow/tfjs-vis": "^1.5.0",
    "@tensorflow/tfjs": "^3.3.0",
    "@tensorflow/tfjs-core": "^3.3.0",
    "@tensorflow/tfjs-backend-wasm": "^3.3.0",
    "@tensorflow/tfjs-backend-webgl": "^3.3.0",
grep @tensorflow node_modules/@tensorflow/tfjs-vis/package.json
  "name": "@tensorflow/tfjs-vis",
    "@tensorflow/tfjs-core": "3.0.0",
    "@tensorflow/tfjs-layers": "3.0.0",
    "@tensorflow/tfjs-backend-webgl": "3.0.0",
    "@tensorflow/tfjs-core": ">= 1.0.0"
```

**Comment by vladmandic** [comment]
@rthadur @lina128 @pyu10055 

I just checked and this is STILL broken in 3.4.0. And it's a regression since `tfjs` 3.1.0
This is a completely TRIVIAL fix and it's still lingering - `tfjs-vis` simply has not updated it's dependencies since tfjs 3.0.0

**Comment by jinjingforever** [comment]
Will publish a new version of the tfjs-vis now

**Comment by jinjingforever** [comment]
Published tfjs-vis 1.5.1 with tfjs 3.5.0 dependencies. Thanks!
